### DISCUSSION TITLE
How to query with case ignored through Lucene

### DISCUSSION THREAD

**Discussion by hyzx86** [OrchardCMS/OrchardCore#11998] [Q&A] [answered]
I have a type that needs to be passed to the Lucene query template through URL parameters
Users may use upper case letters or lower case letters
At present, fuzzy is used, but it will fuzzy match. I'm afraid it will match the wrong result.

For example
input : customer
output: 
customers
Customer
customer
Cusomers

So I want to ignore case on the premise of full word matching

Does anyone know what to do?

Thank you.

**Comment by Skrypt** [comment, marked as answer]
You can create a custom Analyzer for your own usage. But I believe that the standard analyzer should already do it.
Maybe I'm wrong I need to test it. Else, look at what I did with the FrenchAnalyzer on my repositories as an example of a custom Analyzer.

**Reply by hyzx86** [reply]
Is it possible for us to implement `case_insensitive` in regexp filter?

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-regexp-query.html

![image](https://user-images.githubusercontent.com/15613121/178437479-804a9dc4-5f31-42dc-bbb4-126e74f45f70.png)

**Reply by Skrypt** [reply]
Sure, but using this may decrease the query performance vs indexing everything with a lowercase filter.

**Comment by xtomas** [comment]
I think you can downcase searched text parameter in query definition using liquid filter downcase `{{ searchedText | downcase }}`. It works for me without any addtitional backend code.

**Reply by Skrypt** [reply]
Yes and no if the goal is to index the original text and use a proper filter. This way you can have a case sensitive and a non case sensitive search.
